core/templates/dev/head/pages/admin-page/navbar/admin-navbar.directive.ts

```diff
@@ -40,12 +40,24 @@ angular.module('oppia').directive('adminNavbar', [
       controller: ['UserService', function(UserService) {
         var ctrl = this;
         ctrl.ADMIN_TAB_URLS = ADMIN_TAB_URLS;
-        ctrl.showTab = AdminRouterService.showTab;
-        ctrl.isActivitiesTabOpen = AdminRouterService.isActivitiesTabOpen;
-        ctrl.isJobsTabOpen = AdminRouterService.isJobsTabOpen;
-        ctrl.isConfigTabOpen = AdminRouterService.isConfigTabOpen;
-        ctrl.isRolesTabOpen = AdminRouterService.isRolesTabOpen;
-        ctrl.isMiscTabOpen = AdminRouterService.isMiscTabOpen;
+        ctrl.showTab = function() {
+          return AdminRouterService.showTab();
```

Comment:
Any do we need this change?

Reply:
Yeah, this is important. So, the reason is the javascript context binding. When, you call it using $ctrl.showTab = AdminRouterService.showTab the "this" in showTab function is replaced by this of controller which does not have currentTab as a property then. You could try printing console.log(this) in the showTab function. Let me know if you want more information on this. This is a good description for the same: https://javascript.info/object-methods#this-is-not-bound

Reply:
Yeah, "this" is important. :3

I see, thanks for the explanation. Should we hard bind the service? (that won't be a good idea though)

OR can we do `ctrl.adminRouterService = AdminRouterService;` as we are using almost all the methods of the service in the directive, right?

@kevinlee12, do you have any suggestion? (I'm fine with the existing structure as well.)

Reply:
I'm not sure how we are using it currently, but I'll change it to whatever the best practice would be, while I migrate the directives. Thanks~